### PR TITLE
Add Mega Don Quijote

### PR DESCRIPTION
**Pull request by sguinetti**
This pull request adds Mega Don Quijote supermarket. Sources: [YouTube video](https://www.youtube.com/watch?v=k-9LgZpIbgI) and [official website](https://www.donki.com/mega/).

### CODE REVIEW

**Comment by bhousel** [comment]
It looks like a good addition, but we need to fix the `matchNames`.  That should just contain alternate names.  There is a different property called `matchTags` that contains the alternate tags.

I know it's a bit confusing, but I tried to improve the documentation here:  https://github.com/osmlab/name-suggestion-index/blob/master/CONTRIBUTING.md#matchnamesmatchtags

**Comment by sguinetti** [comment]
Ok. I checked that this tag is different from the original Don Quixote variety store. Now I see the implemented matchTags and the tips makes it much better. Thanks for the note.

File: brands/shop/supermarket.json [hunk 1]
```diff
@@ -2617,6 +2617,31 @@
...
+      "brand": "ドン・キホーテ",
+      "brand:en": "Don Quijote",
+      "brand:ja": "ドン・キホーテ",
+      "brand:wikidata": "Q1185381",
+      "brand:wikipedia": "ja:ドン・キホーテ (企業)",
+      "name": "MEGAドン・キホーテ",
+      "name:en": "MEGA Don Quijote",
+      "name:ja": "MEGAドン・キホーテ",
+      "opening_hours": "24/7",
+      "shop": "general",
```
Comment (bhousel) [inline brands/shop/supermarket.json:2639]: Should be like this so it matches the `shop/supermarket` above..
```suggestion
      "shop": "supermarket",
```

Comment (sguinetti) [reply brands/shop/supermarket.json]: Fixed

**Comment by bhousel** [comment]
Thanks, I think this looks ok :+1:

**Pull request merged**